### PR TITLE
[Feature]Blocking task UI and Doc

### PR DESCRIPTION
**Pull request by mgsky1**
## Purpose of the pull request

This PR adds blocking task UI and documentation. The blocking task node backed code has been merged in PR [#6272](https://github.com/apache/dolphinscheduler/pull/6272).

## Brief change log

- add blocking task icon
- add blocking task configuration pannel
- add operations (continue or re-run) when the workflow has been blocked
- add blocking task documentation in Chinese and English

## Screen Shots
blocking_task01

blocking_task02

blocking_task03

blocking_task04

## Verify this pull request

Test cases:
- normal: i.e. Blocking node has upstream and downstram nodes.
- blocking node in the begining of workflow.
- blocking node in the end of workflow.

They all work fine.

## Discuss

- When config blocking node, it has a alert option. In my view, it is better to use switch plugin, but I do not know how to set switch plugin value when the switch turning on or off.
- As for the operation after the workflow is blocked. I think re-run and continue execution is rational. When I testing the operation: continuing from faild node, this operation will skip those node which has been executed successfully. It means that blocking node will lose its function. So I disable this operation.

---

## 此PR的目的

为了完善阻断类任务的开发，后端代码已经在PR[#6272](https://github.com/apache/dolphinscheduler/pull/6272)中被合并

## 变更日志

- 阻断类任务图标
- 阻断节点配置面板
- 当工作流被阻断时，提供重跑和继续执行，两种选择
- 增加阻断节点中英文档

## 截图
blocking_task01

blocking_task02

blocking_task03

blocking_task04

## 测试

主要测试了以下几种情况
- 正常模式：阻断节点上下游都有任务
- 阻断节点在工作流头部（没有上游节点）
- 阻断节点在工作流尾部（没有下游节点）

它们都能正常工作

## 讨论

- 在阻断节点的配置中，有一个告警选项，我觉得将其作为switch开关更合适，但是我不知道如何给这个switch开关赋值，从代码来看，新的ui将配置项的每一条都抽出进行了高度封装，返回`IJsonItem`数组，不知道这种场景下，要用`IJsonItem`的什么字段给开关赋值。从现有的代码参考看，如果是在节点配置面板中使用了switch开关，那么开启开关后，会弹出更详细的设置信息；另外一处是在工作流空跑时出现的，那个场景是直接用HTML标签搞定的。
- 对于工作流被阻断后的操作，我觉得继续执行和重跑是合理的。对于重新执行失败节点，在测试中发现，这种模式会跳过已经执行成功的节点，这会导致阻断节点的失效，因此不启用。

### CODE REVIEW

**Comment by SbloodyS** [comment]
That's a cool~

File: docs/docs/zh/guide/task/blocking.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+![blocking-task03](../../../../img/tasks/demo/blocking_task04.png)
+
+![blocking-task03](../../../../img/tasks/demo/blocking_task03.png)
+
+这时，你可以选择：**重跑**或者**继续执行**工作流
+
+如果你有配置告警相关组件（例如email），那么将会收到一封告警邮件
+
+##
```
Comment (zhongjiajie) [inline docs/docs/zh/guide/task/blocking.md]: It seems that is unnecessary adding

File: docs/docs/en/guide/task/blocking.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+- Alert on blocking：Whether to send alerts to relevant users when a workflow is blocked.
+- Add pre task check condition：i.e. **custom parameter**. One or more upstream tasks can be selected for blocking node to build workflow blocking logic.
+    - Add the upstream dependency：Use the first parameter to choose task name, and the second parameter for status of the upsteam task.
+    - Upstream task relationship：we use `and` and `or`operators to handle complex relationship of upstream when multiple upstream tasks for Conditions task.
+
+## Example
+
+This sample demonstrates the operation of the Condition task by using the [Shell](shell.md) task and [Python](python.md) task.
+
+### 1. Create Workflow
```
Comment (zhongjiajie) [inline docs/docs/en/guide/task/blocking.md]: should avoid using number in title, the leading number of `#` represent the level
```suggestion
### Create Workflow
```

**Review by zhongjiajie — COMMENTED** [review]
backend and docs LGTM, @devosend please task a look at the frontend code

**Comment by mgsky1** [comment]
I am very very sorry for updating this PR so late because of the busy work and training. I see projects iterating rapidly and the back-end code for the blocking node is still being maintained. I sincerely appreciated. 
@zhongjiajie @caishunfeng

**Comment by EricGao888** [comment]
> I am very very sorry for updating this PR so late because of the busy work and training. I see projects iterating rapidly and the back-end code for the blocking node is still being maintained. I sincerely appreciated. @zhongjiajie @caishunfeng

@mgsky1 No worries, we are glad that you could follow up with this PR. The CI failed because of formatting errors, could u plz run `mvn spotless:apply` under the root directory so that `spotless` will automatically fix those errors for you?

**Comment by mgsky1** [comment]
> > I am very very sorry for updating this PR so late because of the busy work and training. I see projects iterating rapidly and the back-end code for the blocking node is still being maintained. I sincerely appreciated. @zhongjiajie @caishunfeng
> 
> @mgsky1 No worries, we are glad that you could follow up with this PR. The CI failed because of formatting errors, could u plz run `mvn spotless:apply` under the root directory so that `spotless` will automatically fix those errors for you?

ok，I will run this command later today~

**Comment by mgsky1** [comment]
There are still some errors occured in the Actions. I will fixed them later, maybe in this weekend

**Comment by mgsky1** [comment]
I fixed some error in code so that the Actions could be executed. I also clean the commits. The commit logs before are so terrible.@EricGao888

**Comment by caishunfeng** [comment]
Hi @mgsky1 It's great to follow up the blocking task feature :+1: 
Please resolve the conflicts.

**Comment by caishunfeng** [comment]
@devosend @Amy0104 @songjianet Do you have time to review this pr?

**Review by devosend — APPROVED** [review]
frontend LGTM

**Comment by songjianet** [comment]
Please resolve the conflict. @mgsky1

**Pull request closed without merging**